nwbinspector/checks/tables.py

```diff
@@ -183,3 +183,18 @@ def check_table_values_for_dict(table: DynamicTable, nelems: int = 200):
                 if is_string_json_loadable(string=string):
                     message += " This string is also JSON loadable, so call `json.loads(...)` on the string to unpack."
                 yield InspectorMessage(message=message)
+
+
+@register_check(importance=Importance.BEST_PRACTICE_SUGGESTION, neurodata_type=DynamicTable)
+def check_col_not_nan(table: DynamicTable, nelems: int = 200):
+    """Check if all of the values in a single column of a table are NaN."""
+    for column in table.columns:
+        if not hasattr(column, "data") or isinstance(column, VectorIndex) or isinstance(column.data[0], str):
+            continue
+        subindex_selection = np.unique(np.round(np.linspace(start=0, stop=column.shape[0] - 1, num=nelems)).astype(int))
```

Comment:
what's going on here? Why not just slice with nelems?

Reply:
hmm ok. This has the downside of likely needing to access every chunk, so for chunked data at this point you are essentially reading the entire column

Reply:
Would have to be a huge column (>125,000 rows of float elements) to have multiple chunks, though, but yes what you say is true - moreso for the TimeSeries one, but it's even more important to do it there since 200 otherwise consecutive elements of potentially high-frequency data isn't nearly enough to safely determine the logic.

Reply:
How about for the TimeSeries one, we _first_ check if the first `nelemes` consecutive elements are all `NaN`, and if they are then check it again using the uniform sampling over the entire series? I do understand that if streaming on a long TimeSeries, the uniform sampling might even pull `nelems` number of total chunks, but this would make it so that we only request that much data if there is reason to do so.

Reply:
sounds fine

Reply:
OK, added here - adding to the other PR now